timon/state.py

```diff
@@ -115,10 +130,28 @@ def items(self):
             yield value
 
     def as_dict(self):
-        """ returns queue as a jsonable list """
-        rslt = dict(heap=self.heap,  sched_dict=self.sched_dict)
+        """
+        returns queue as a jsonable dict.
+        Makes copy of heap and sched dict (caution: actually, no need of a deep
+        copy so a simple copy is made)
+        """
+        copied_heap = self.heap.copy()
+        copied_heap.sort()
+        rslt = dict(heap=copied_heap, sched_dict=self.sched_dict.copy())
         return rslt
 
+    def get_probe_info_in_heap(self, probename):
```

Comment:
you insist using the word 'heap' (which is an implementation detail) in public interfaces

is this complicated name really necessary.

if you write `queue.get_probe(probename)`
|
then I don't see what added information you get if you write

`queue.get_probe_from_queue(probename)`

Reply:
you might perhaps write `get_probe_n_schedule()`